sklearn/linear_model/_glm/glm.py

```diff
@@ -279,7 +298,9 @@ def fit(self, X, y, sample_weight=None):
                     family.__class__.__name__
                 )
             )
-        # TODO: if alpha=0 check that X is not rank deficient
+        # check that matrix is full rank when alpha = 0.
+        if np.max(self.alpha) == 0.0 and np.linalg.matrix_rank(X) < X.shape[1]:
+            raise ValueError("X must be full rank when alpha = 0.")
```

Comment:
Please revert. This should be a seperate PR. Meanwhile, I'm not even convinced anymore that this is necessary.

Reply:
reverted